src/DeviceInterface/InterfacePrototype.js

```diff
@@ -264,21 +265,40 @@ class InterfacePrototype {
     }
 
     /**
-     * @param {Map<HTMLElement, import("../Form/Form").Form>} forms
+     * Initializes a global focus event handler that handles iOS keyboard and autocomplete functionality
+     * @param {Map<HTMLElement, import("../Form/Form").Form>} forms - Collection of form objects to monitor
+     * @returns {void}
      */
     initGlobalFocusHandler(forms) {
-        window.addEventListener(
-            'focus',
-            (e) => {
-                const isAnyFormAutofilling = [...forms.values()].some((form) => form.isAutofilling);
-                const form = [...forms.values()].find((form) => form.hasFocus());
-                const targetElement = pierceShadowTree(e);
-                if (!isAnyFormAutofilling && this.globalConfig.isIOS && targetElement && !(targetElement instanceof Window)) {
-                    this.attachKeyboard({ device: this, form, element: targetElement });
-                }
-            },
-            true,
-        );
+        const { setAutocompleteOnIdentityField } = initAutocompleteApi();
+
+        window.addEventListener('focus', this._handleFocusEvent.bind(this, forms, setAutocompleteOnIdentityField), true);
+    }
+
+    /**
+     * Handles focus events for iOS devices
+     * @param {Map} forms - Collection of form objects
+     * @param {Function} setAutocompleteOnIdentityField - Function to set autocomplete attributes
+     * @param {FocusEvent} e - The focus event
+     * @private
+     */
+    _handleFocusEvent(forms, setAutocompleteOnIdentityField, e) {
+        // Check if any form is currently autofilling
+        const isAnyFormAutofilling = [...forms.values()].some((form) => form.isAutofilling);
+        if (isAnyFormAutofilling) return;
+
+        const targetElement = pierceShadowTree(e);
+
+        if (!this.globalConfig.isIOS || !targetElement || targetElement instanceof Window) return;
+        const form = [...forms.values()].find((form) => form.hasFocus());
+
+        if (this.settings.featureToggles.input_focus_api) {
+            this.attachKeyboard({ device: this, form, element: targetElement });
+        }
+
+        if (this.settings.featureToggles.autocomplete_attribute_support) {
+            setAutocompleteOnIdentityField(targetElement);
+        }
```

Comment:
Why aren't we encapsulating all this stuff inside a focus API of sorts? Seems more appropriate to have a single focus object that includes everything, rather than having some things inline in the prototype here and some others in the autocomplete API. The prototype would just check the feature flags and initialize the API as needed. We do this for other similar features.